DQM/HcalTasks/python/OfflineSourceSequence_pp.py

```diff
@@ -32,6 +34,17 @@
     recHitPreRecoTask +
     rawTask )
 
+hcalOnlyOfflineSourceSequenceGPU = cms.Sequence(
+    digiTask +
+    recHitTask +
+    rawTask +
+    DQMHcalGPUComparisonTask
+)
+
+from Configuration.ProcessModifiers.gpu_cff import gpu
+gpu.toReplaceWith(hcalOnlyOfflineSourceSequence, hcalOnlyOfflineSourceSequenceGPU)
```

Comment:
Using the `gpu` modifier to run the GPU validation task results in a request for the `@cpu` and the `@cuda` collections by the DQM module when giving the modifier. This means that whenever we activate the GPU code we automatically run the CPU code as well. So instead of a performance improvement we will have a penalty from running both, GPU and CPU code.

As discussed in https://github.com/cms-sw/cmssw/pull/36742#issuecomment-1043135863 a solution for this is to create a separate modifier `gpuValidationHcal` (as done for ECAL here: https://github.com/cms-sw/cmssw/pull/36742/files#diff-7bb29b5f06ec4140854884cfb7801d25c2c3a3b29799af42681720270eda27e9R1), use it here instead of `gpu`, and add the modifier to the `gpuValidation` ModifierChain here https://github.com/cms-sw/cmssw/pull/36742/files#diff-07b659e3b734a1800bf9d7285a7e693eaf9e956baa074e038b0c702e112997edR10

The `gpuValidation` process modifier can then be given to activate the GPU validation. Or the `gpuValidationHcal` alone in combination with the `gpu` modifier if only the HCAL validation is desired for some reason.

Since #36742 is not merged yet this will lead to a conflict that will have to be resolved once it is in.